image_provisioner/playbooks/roles/docker-config/tasks/main.yaml

```diff
@@ -12,16 +21,61 @@
 - name: install docker
   yum: name=docker state=latest
 
-- name: check for existing PV
-  command: pvdisplay
-  register: pv_result
+- block:
+   # should be used until the mirrors are fixed to install right container-selinux package
+   - name: remove existing container-selinux package
+     yum: name=container-selinux state=absent
+
+   - name: install container-selinux-1.12.5-14
+     yum: name=container-selinux-1.12.5-14.el7 state=latest
+
+   - name: create a xfs filesystem
+     shell: mkfs.xfs -f /dev/xvdb
+
+   - name: update fstab
+     shell: echo "/dev/xvdb /var/lib/docker                    xfs   defaults        0 0" >> /etc/fstab
+
+  when: docker_storage_driver == "overlay"
+
+- block:
+   - name: restart machine
```

Comment:
upgrade_kernel=false
update_kernel=false
docker_storage_driver=devicemapper

TASK [docker-config : restart machine] *****************************************
Thursday 16 February 2017  14:48:07 -0500 (0:00:00.019)       0:01:19.381 ***** 
ok: [192.168.124.55]

TASK [docker-config : waiting for server to come back] *************************
Thursday 16 February 2017  14:48:08 -0500 (0:00:01.191)       0:01:20.572 ***** 
ok: [192.168.124.55 -> localhost]

TASK [docker-config : get current kernel version] ******************************
Thursday 16 February 2017  14:53:23 -0500 (0:05:15.240)       0:06:35.813 ***** 
changed: [192.168.124.55]

TASK [docker-config : check if current kernel version is 3.10.0-517 or later] **
Thursday 16 February 2017  14:53:24 -0500 (0:00:00.378)       0:06:36.192 ***** 
fatal: [192.168.124.55]: FAILED! => {"changed": true, "cmd": "rpmdev-vercmp 3.10.0-563.el7.x86_64 3.10.0-517 > /dev/null", "delta": "0:00:00.059143", "end": "2017-02-16 14:53:23.585054", "failed": true, "rc": 11, "start": "20
17-02-16 14:53:23.525911", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}
...ignoring

TASK [docker-config : exit if kernel version is outdated] **********************
Thursday 16 February 2017  14:53:24 -0500 (0:00:00.171)       0:06:36.363 ***** 
skipping: [192.168.124.55]